docs/configuration.asciidoc

```diff
@@ -1,57 +1,50 @@
 [[configuration]]
 == Configuration
 
-Configure the agent with `AgentConfiguration` in the agent's `start()` function.
+Configure the agent with `AgentConfigBuilder` passing the `AgentConfiguration` to the `start` function.
 
 // some config example that preferably is correct unlike mine
 [source,swift]
 ----
-let config =  AgentConfiguration()
-config.collectorHost = "192.168.0.1"
-config.collectorPort = 8200
-config.collectorTLS = true
-config.secretToken = "<secret token>"
+let config = AgentConfigBuilder()
+                .withServerUrl(URL(string: "http://localhost:8200"))
+                .withSecretToken("<Token>")
+                .build()
+
 Agent.start(with:config)
 ----
 
+The `AgentConfigBuilder` can be configured with the following functions
+
 [discrete]
 [[configuration-options]]
 === Configuration options
 
-
 [discrete]
-[[collectorHost]]
-==== `collectorHost`
-
-* *Type:* String
-* *Default:* `127.0.0.1`
-// * *Env:* ``
-
-The APM Server host.
+[[withServerUrl]]
+==== `withServerUrl`
 
+* *Type:* URL
+* *Default:* `http://127.0.0.1:8200`
 [discrete]
-[[collectorPort]]
-==== `collectorPort`
-
-* *Type:* Int
-* *Default:* `8200`
-// * *Env:* ``
-
-The APM Server port.
+[[withSecretToken]]
```

Comment:
👋 @bryce-b  it looks like the changes in this PR broke the docs build. There are several places across APM docs that link to the [`secretToken` section of configuration page](https://www.elastic.co/guide/en/apm/agent/swift/current/configuration.html#secretToken), but here that section ID was changed to `withSecretToken` causing broken links. Can you change this section ID to `secretToken`?

```suggestion
[[secretToken]]
```

(I think this is probably faster than the alternative, which is to fix all broken links in apm-server, which would include updating the links that appear in https://www.elastic.co/guide/en/apm/guide/current/secret-token.html and https://www.elastic.co/guide/en/apm/guide/current/secret-token-legacy.html on all branches all the way back to `7.16`.)